apps/ejabberd/src/ejabberd_auth.erl

```diff
@@ -246,12 +246,18 @@ set_password(User, Server, Password) ->
 do_set_password(LUser, LServer, _) when LUser =:= error; LServer =:= error ->
     {error, invalid_jid};
 do_set_password(LUser, LServer, Password) ->
-    lists:foldl(
-      fun(M, {error, _}) ->
-              M:set_password(LUser, LServer, Password);
-         (_M, Res) ->
-              Res
-      end, {error, not_allowed}, auth_modules(LServer)).
+    case is_user_exists(LUser, LServer) of
+        true ->
+            lists:foldl(
+                fun(M, {error, _}) ->
+                    M:set_password(LUser, LServer, Password);
+                   (_M, Res) ->
+                       Res
+                end, {error, not_allowed}, auth_modules(LServer));
+        false ->
+            {error, user_not_exists}
```

Comment:
`does_user_exists` is in backend modules only (at least now). In `ejabberd_auth` we can use `do_does_user_exist` which is local function and it's called form `is_user_exists`